src/pretix/base/services/cart.py

```diff
@@ -1132,9 +1143,15 @@ def _perform_operations(self):
                     voucher_available_count = min(voucher_available_count, vouchers_ok[op.voucher])
 
                 if quota_available_count < 1:
-                    err = err or error_messages['unavailable']
+                    err = err or error_messages['unavailable_listed']
```

Comment:
If I understand correctly, if we have both errors in the same operation (unavailable_listed or in_part_listed), we display whatever error message occurs first, but with all affected products? This might be confusing... OTOH I guess it's a pretty narrow edge case so it's probably okay to ignore.

Reply:
Well... yes. A list of error messages would probably be cleaner.